help needed draft (#636)
ropensci/roweb3: content/blog/2023-10-03-help-needed-issues/index.md

---

> title: 'Attract (and retain) new contributors with "help wanted" issues'

Reviewer: Aside from "thanking" your contributors, the post doesn't really deal with retaining contributions, and I'm not sure it's necessary
```suggestion
title: 'Attract contributors with "help wanted" issues'
```

author:
  - Maëlle Salmon
  - Yanina Bellini Saibene
date: '2023-10-03'
slug: help-wanted
featured: true 
tags:
  - community
  - packages
  - welcome

>   - maintenance

Reviewer: ```suggestion
  - maintenance
  - contributors
```

Acknowledge contributions [generously](https://devguide.ropensci.org/collaboration.html?q=generous#attributions).

## Don't despair

Even if you wrote an excellent issue, it might just not get picked up.
In that case, alternatives or complements might be to broadcast it again, ask for general tips to your fellow maintainers, try and get funding (therefore time, either yours or an external contractor's?) for your maintenance efforts.
See for instance the [R Consortium's two yearly calls for proposals](https://www.r-consortium.org/all-projects/call-for-proposals#Rstats).

Even if no one tackles the issue in the end, it might be good to have thought about the possible resolution, which can help you solve your problem. 
Also, a well-documented issue is a way to document your decisions about your software transparently and can help future users and contributors understand the reasons for your choices.

## Conclusion

> Opening "help wanted" issues can be a way to grow your package contributors' community.

Reviewer: ```suggestion
Opening "help wanted" issues can be a way to grow your package's contributor community.
```

They might be a better door to contribution than less specific issues with a laundry list of needed tasks, as they're less overwhelming.

> Contributors might fix one "help wanted" issue and then leave, or go ahead and solve more of them.

Reviewer: ```suggestion
Contributors might fix one "help wanted" issue and then leave, or go ahead and solve more of them.
Sometimes a conversation in the comments can help you find a solution even if a contributor doesn't submit a PR.
```

By listing them in your issue tracker, 

- you show curious users that they're on your mind, 
- you can organize your own work... 
- and you can receive external help if you signal it's ok, with the "help wanted" label.

## Polish your issue and contributing guide

Once you have a topic in mind, make the issue title and text as clear as possible.
Even if you are the one fixing the issue in the end, future you will be happy you didn't jot down an undecipherable comment.
Link to resources as needed, indicate context.
Approach the issue writing as you approach the issue writing in a repo that isn't yours: challenge description, desired outcome, trade-offs, etc.

Beyond efforts in the individual issue, it's crucial to have a [contributing guide](https://devguide.ropensci.org/collaboration.html#contributing-guide) underlining anything that is good to know about contributing to your package: tooling used, style and design preferences.
Do not duplicate external resources, point to them instead.
Do not overload or scare off contributors by having too strict requirements: you can probably finish off PRs yourself, or teach them little by little. 

> 

Reviewer 2: I shared a paper on Slack about contributing guides in OSS: https://fron.ch/preprint_fse.pdf. They have this tool to check your CONTRIBUTING file in a GitHub repo: https://contributing.streamlit.app/, perhaps mention it? Or would be too much?

Then you could also share it within your own networks: rOpenSci slack workspace, your Mastodon account, a local R User Group communication channel, etc.

Some tools or people browse public GitHub issues labelled "help wanted", for [Hacktoberfest](https://hacktoberfest.com/) for instance if you add the ["hacktoberfest" topic to your repository](https://hacktoberfest.com/participation/), but you can't count on someone's discovering your tiny issue in that sea of issues.

## Thank contributors

If someone answers in the issue or open a PR, try to be somewhat responsive.
Check your settings allow you to be notified of issue comments and PRs, you might need to [watch](https://docs.github.com/en/account-and-profile/managing-subscriptions-and-notifications-on-github/managing-subscriptions-for-activity-on-github/viewing-your-subscriptions) your repository.

Acknowledge contributions [generously](https://devguide.ropensci.org/collaboration.html?q=generous#attributions).

## Don't despair

Even if you wrote an excellent issue, it might just not get picked up.

> In that case, alternatives or complements might be to broadcast it again, ask for general tips to your fellow maintainers, try and get funding (therefore time, either yours or an external contractor's?) for your maintenance efforts.

Reviewer 2: Perhaps we can link to the RConsortium ISC Call for grants?

## Conclusion

Opening "help wanted" issues can be a way to grow your package contributors' community.
They might be a better door to contribution than less specific issues with a laundry list of needed tasks, as they're less overwhelming.
Contributors might fix one "help wanted" issue and then leave, or go ahead and solve more of them.

As a contributor, always comment in an issue before tackling it, to ensure it's still up-to-date, and that no one else is preparing a duplicate PR right now!
How vexing would that be to work for nothing.

> They might appear as a potential loss of time if no one tackles the issue in the end, but it might be good to have thought about the issue resolution even if you solve it yourself in the end.

Reviewer 2: What do you think about giving a positive tone to this phrase? Something like this: 
Even if no one tackles the issue in the end, it might be good to think about the resolution, which can help you solve it. Also, a well-documented issue is a way to document your decisions about your software transparently and can help future users and contributors understand the reasons for your implementations.
